services/QuillLMS/engines/evidence/spec/controllers/evidence/feedback_controller_spec.rb

```diff
@@ -321,327 +320,5 @@ module Evidence
-        before do
-          create(:evidence_feedback, rule_id: opinion_rule.id, text: 'lorem ipsum')
-        end
-
-        it 'should return a valid json response' do
-          post :opinion, params: incoming_payload, as: :json
-
-          expect(JSON.parse(response.body)).to eq({
-            'concept_uid' => 'xyz',
-            'feedback' => 'lorem ipsum',
-            'feedback_type' => 'opinion',
-            'optimal' => false,
-            'highlight' => [
-              { 'type' => 'response', 'text' => 'someText', 'character' => 0 }
-            ],
-            'hint' => { 'id' => opinion_hint.id, 'explanation' => opinion_hint.explanation, 'image_link' => opinion_hint.image_link, 'image_alt_text' => opinion_hint.image_alt_text, 'rule_id' => opinion_hint.rule_id },
-            'labels' => '',
-            'rule_uid' => example_rule_uid.to_s
-          })
-        end
-      end
-
-    end
-
-    describe '#prefilter' do
-      let!(:profanity_rule) do
-        create(:evidence_rule, rule_type: 'prefilter', uid: 'fdee458a-f017-4f9a-a7d4-a72d1143abeb')
-      end
-
-      context 'profanity violation' do
-        it 'should return correct rule uid' do
-          post "prefilter", :params => ({
-            :entry => "nero was an ahole",
-            :prompt_id => prompt.id,
-            :session_id => 1,
-            :previous_feedback => ([])
-          }), :as => :json
-          expect(response.status).to eq 200
-          parsed_response = JSON.parse(response.body)
-          expect(parsed_response['optimal']).to be false
-          expect(parsed_response['rule_uid']).to eq profanity_rule.uid
-        end
-      end
-
-      context 'no violation' do
-        it 'should return successfully' do
-          post "prefilter", :params => ({
-            :entry => "the quick brown fox",
-            :prompt_id => prompt.id,
-            :session_id => 1,
-            :previous_feedback => ([])
-          }), :as => :json
-          expect(response.status).to eq 200
-          parsed_response = JSON.parse(response.body)
-          expect(parsed_response['optimal']).to be true
-          expect(parsed_response['rule_uid']).to eq PrefilterCheck::OPTIMAL_RULE_UID
-        end
-      end
-    end
-
-    context 'should plagiarism' do
-      it 'should return successfully' do
-        post("plagiarism", :params => ({ :entry => "No plagiarism here.", :prompt_id => prompt.id, :session_id => 1, :previous_feedback => ([]) }), :as => :json)
-        parsed_response = JSON.parse(response.body)
-        expect(parsed_response["optimal"]).to be true
-      end
-
-      it 'should return 404 if prompt id does not exist' do
-        post("plagiarism", :params => ({ :entry => "No plagiarism here.", :prompt_id => 100, :session_id => 1, :previous_feedback => ([]) }), :as => :json)
-        expect(response.status).to(eq(404))
-      end
-
-      it 'should return successfully when there is plagiarism that matches the first plagiarism text string' do
```

Comment:
I think these tests should be ported to the `create` endpoint. You can isolate plagiarism by stubbing the check to only run plagiarism. `stub_const("Evidence::Check::ALL_CHECKS", [Check::Plagiarism])`